src/main/java/org/dependencytrack/tasks/vulnerabilitypolicy/S3Client.java

```diff
@@ -0,0 +1,57 @@
+package org.dependencytrack.tasks.vulnerabilitypolicy;
+
+import alpine.Config;
+import alpine.common.logging.Logger;
+import io.minio.MinioClient;
+import org.dependencytrack.common.ConfigKey;
+
+public class S3Client {
+    private final String bundleToFetch;
+    private final String s3AccessKey;
+    private final String s3SecretKey;
+    private final String s3BucketName;
+    private MinioClient minioClient;
+
+    private String bundleFetchEndpoint;
+
+    private final String region;
+
+    public String getBundleToFetch() {
+        return bundleToFetch;
+    }
+
+    public String getS3BucketName() {
+        return s3BucketName;
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(S3Client.class);
+
+    public S3Client() {
+        this.bundleFetchEndpoint = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_BUNDLE_URL);
+        this.bundleToFetch = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_BUNDLE_NAME);
+        this.s3AccessKey = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_ACCESS_KEY);
+        this.s3SecretKey = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_SECRET_KEY);
+        this.s3BucketName = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_BUCKET_NAME);
+        this.region = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_REGION);
+        if (bundleToFetch == null || s3BucketName == null || s3SecretKey == null || s3AccessKey == null) {
+            LOGGER.error("The required parameters for S3 file access have not been provided. Please check");
+        }
```

Comment:
Why not throw an `IllegalArgumentException` here? The caller would otherwise never know that this client has no way of ever working correctly.